### DISCUSSION TITLE
Add support for writing .dds files with BC7 compression.

### DISCUSSION THREAD

**Discussion by dlemstra** [ImageMagick/ImageMagick#6694] [Contribute]
### Describe the solution you'd like

In the pull request #4126 from @SubhadeepG support for reading `.dds` files with `BC7` compression was added. It would be nice if we could also add support for writing `.dds` images with this compression.

**Comment by GraPhiX-Guru** [comment]
Sorry for causing inconvenience and trouble :( here is a BC7 format image from paint.net
[myskybox_F.zip](https://github.com/ImageMagick/ImageMagick/files/7310393/myskybox_F.zip)

i have to zip it GitHub does not support DDS images

**Comment by dlemstra** [comment]
No worries, I was just pointing out that I prefer to keep talking in a single issue or discussion instead of opening multiple about the same thing.

**Comment by Arnauw** [comment]
It could be very nice if magick could do that, any updates on this ?

**Comment by holzerseb** [comment]
Are there any updates on this @urban-warrior ? I have alot of DDS with BC7 compression and would really love to have Imagemagick support. Thanks!

**Comment by snibgo** [comment]
> I have alot of DDS with BC7 compression ...

The OP says that IM can *read* DDS with BC7 compression, but can't *write* it.

**Comment by zhro** [comment]
Following up on this. I needed to do a bulk operation on over 100,000 .DDS files, too many to do manually, and thought that IM would be perfect for this. However, a lot of operations failed because there was no way to write the changes back to disk in BC7 format.

**Reply by fmw42** [reply]
What is your IM version and platform?  Are you using the most current version of Imagemagick? Sorry, I do not know if IM will write that format. A developer might comment here.

**Reply by GraPhiX-Guru** [reply]
what are the source DDS files and what operation are you trying to do ??

**Reply by GraPhiX-Guru** [reply]
I only ask because i wrote an app that will batch covert file formats
![xconvert](https://github.com/user-attachments/assets/acce8ca9-4e5a-4bdb-bed3-8ed39493d49c)

**Reply by fmw42** [reply]
Please answer my questions above!  Please show your code and post and example input image to process so we can test.

**Comment by zhro** [comment]
This was my mistake. I downloaded the latest ImageMagick but it was not set correctly in my PATH. It appears to work as expected now with BC7 DDS files. Thank you.

**Comment by dlemstra** [comment]
Just to be clear... this discussion is about adding support for writing BC7 files. We now only have support for reading BC7 files.

**Comment by h0lley** [comment]
Confirming DDS compression for BC7 still falls back to DXT5 in ImageMagick 7.1.1-38 (39 doesn't appear have any updates on it, either).

I am not sure how commonly BC7 is used, but my use case pertains to a modern game that uses this compression.

What's more, tooling for both DDS reading and especially writing appears to be very limited for Linux, making this feature in IM highly desirable.

Meanwhile this did the trick for me:

`wine texconv.exe -f BC7_UNORM_SRGB -ft dds -srgbi -m 1 -y in.png out.dds`

[Texconv](https://github.com/microsoft/DirectXTex) is a free tool from MS.
